pkgs/development/libraries/x265/fix-neon-detection.patch

```diff
@@ -0,0 +1,28 @@
+commit 72489cd0a1c229258abe4f20e4fdfd414dfa88da
+Author: rnhmjoj <rnhmjoj@inventati.org>
+Date:   Sun Oct 2 00:15:24 2022 +0200
+
+    Fix NEON detection
+
+diff --git a/cmake/FindNeon.cmake b/cmake/FindNeon.cmake
+index 0062449..9c436d9 100644
+--- a/cmake/FindNeon.cmake
++++ b/cmake/FindNeon.cmake
+@@ -1,10 +1,11 @@
+ include(FindPackageHandleStandardArgs)
+ 
+ # Check the version of neon supported by the ARM CPU
+-execute_process(COMMAND cat /proc/cpuinfo | grep Features | grep neon
+-                OUTPUT_VARIABLE neon_version
+-                ERROR_QUIET
+-                OUTPUT_STRIP_TRAILING_WHITESPACE)
+-if(neon_version)
+-    set(CPU_HAS_NEON 1)
++message(STATUS "Detecting NEON support")
++execute_process(COMMAND sed -n "/Features.* neon/q 1" /proc/cpuinfo
++                RESULT_VARIABLE CPU_HAS_NEON)
++if(CPU_HAS_NEON)
++    message(STATUS "Detecting NEON support - supported")
++else()
++    message(STATUS "Detecting NEON support - not supported" )
```

Comment:
Uh... We probably would wand to set the feature and not look at the ambiant system during cross-compilation.

This change wouldn't "detect" neon support on an x86_64 build platform, because obviously the x86_64 system won't have neon support!

I haven't tested this fresh change, but is it that this patch works around some failure from the previous pipeline here, but not actually change the expected result?

Reply:
> We probably would want to set the feature and not look at the ambiant system during cross-compilation.

Note that this is not used when cross compiling (`-DCROSS_COMPILE_ARM=ON`).

> And even when building natively, looking at the ambiant system is wrong!

Yeah, I only made it less wrong. Before it was completely broken (`execute_process` doesn't execute a shell command but a single program), so it would always find NEON support, which is very wrong if you're building on armv6.

Reply:
Right, so the patch is not a fix for armv6 cross-compilation, but a patch for native builds, is that it?

Reply:
> but a patch for native builds, is that it?

Correct.